pkgs/development/ocaml-modules/facile/default.nix

```diff
@@ -1,18 +1,19 @@
-{ lib, fetchurl, buildDunePackage }:
+{ lib, fetchurl, buildDunePackage, ocaml }:
 
 buildDunePackage rec {
   pname = "facile";
   version = "1.1.4";
 
-  useDune2 = false;
-
   src = fetchurl {
     url = "https://github.com/Emmanuel-PLF/facile/releases/download/${version}/facile-${version}.tbz";
     sha256 = "0jqrwmn6fr2vj2rrbllwxq4cmxykv7zh0y4vnngx29f5084a04jp";
   };
 
   doCheck = true;
 
+  useDune2 = lib.versionAtLeast ocaml.version "4.12";
+  postPatch = lib.optionalString (lib.versionAtLeast ocaml.version "4.12") "dune upgrade";
```

Comment:
```suggestion
  postPatch = lib.optionalString useDune2 "dune upgrade";
```

Reply:
That's much smarter, changed both facile and nonstd